packages/hint-manifest-icons/src/hint.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * @fileoverview Checks usage of Web App Manifest icons
+ */
+import { URL } from 'url';
+import * as getImageData from 'image-size';
+import { HintContext } from 'hint/dist/src/lib/hint-context';
+// The list of types depends on the events you want to capture.
+import { IHint, NetworkData } from 'hint/dist/src/lib/types';
+import {
+    ManifestEvents,
+    ManifestParsed,
+    ManifestImageResource
+} from '@hint/parser-manifest';
+import { debug as d } from '@hint/utils';
+
+import meta from './meta';
+
+const debug: debug.IDebugger = d(__filename);
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+export default class ManifestIconHint implements IHint {
+    public static readonly meta = meta;
+    public constructor(context: HintContext<ManifestEvents>) {
+        /**
+         * Try to see if the `icon` file actually
+         * exists and is accessible.
+         */
+        const iconExists = async (IconPath: string) => {
+            let networkData: NetworkData;
+
+            try {
+                networkData = await context.fetchContent(IconPath);
+            } catch (e) {
+                debug(`Failed to fetch the ${IconPath} file`);
+                const message = `'${IconPath}' could not be fetched (request failed).`;
+
+                context.report(IconPath, message);
+
+                return;
+            }
+            const response = networkData.response;
+
+            if (response.statusCode !== 200) {
+                const message = `'${IconPath}' could not be fetched (status code: ${
+                    response.statusCode
+                }).`;
+
+                context.report(IconPath, message);
+
+                return;
+            }
+
+            let image;
+
+            try {
+                image = getImageData(response.body.rawContent);
+            } catch (e) {
+                if (e instanceof TypeError) {
+                    const message = `'${IconPath}' should be a valid PNG image.`;
+
+                    context.report(IconPath, message);
+                } else {
+                    debug(`'getImageData' failed for '${IconPath}'`);
+                }
+
+                return;
+            }
+
+            // Check if the image is a PNG.
+            if (image.type !== 'png') {
+                const message = `'${IconPath}' should be a PNG image but found be a
```

Comment:
What about:
```suggestion
                const message = `Manifest icon should be of type "png" but found "${image.type}"`;
```

We are already providing `iconPath` in `report` so no need to put it in the message too.

Reply:
Yeah, that makes sense.